Unity-Technologies/Graphics
# [10.x.x] Fix incorrect intermediate resolution when rendering to RT

### Purpose of this PR
Fixes [case 1261287](https://issuetracker.unity3d.com/product/unity/issues/guid/1261287/). The repro project renders into slices of a render texture by modifying the `Camera.rect` property. When we then create intermediate textures in URP, we use the target texture resolution, but that doesn't account for the `rect`. This PR instead uses `Camera.pixelWidth` and `Camera.pixelHeight` when rendering to a render texture.

---
### Testing status

**Manual Tests**: What did you do?

**Automated Tests**: We already have 119_CameraToRTWithViewportRect in UniversalGraphicsTest to test this, but the threshold was set too high to fail. I've lowered the threshold, and modified the scene to cause a bigger change when things are wrong.

The modified scene looks like this without the fix:
![119_CameraToRTWithViewportRect](https://user-images.githubusercontent.com/123919/89034678-c0d4f580-d339-11ea-919e-c2330c20f35f.png)

With the fix it looks like this:
![119_CameraToRTWithViewportRect expected](https://user-images.githubusercontent.com/123919/89034690-c7636d00-d339-11ea-8dfc-07d00a8d3ce4.png)

**Yamato**: https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/urp%252Ffix-1261287

---
### Comments to reviewers
There is an issue with UniversalGraphicsTest on Win_DX11_playmode_XR that I am not sure what is caused by.
